shell/platform/common/accessibility_bridge_unittests.cc

```diff
@@ -337,5 +337,30 @@ TEST(AccessibilityBridgeTest, SliderHasSliderRole) {
   EXPECT_EQ(root_node->GetData().role, ax::mojom::Role::kSlider);
 }
 
+TEST(AccessibilityBridgeTest, canSetCheckboxChecked) {
+  std::shared_ptr<AccessibilityBridge> bridge =
+      std::make_shared<AccessibilityBridge>(
+          std::make_unique<TestAccessibilityBridgeDelegate>());
+  FlutterSemanticsNode root;
+  root.id = 0;
+  root.label = "root";
+  root.hint = "";
+  root.value = "";
+  root.increased_value = "";
+  root.decreased_value = "";
+  root.child_count = 0;
+  root.custom_accessibility_actions_count = 0;
+  root.flags = static_cast<FlutterSemanticsFlag>(
+      FlutterSemanticsFlag::kFlutterSemanticsFlagHasCheckedState |
+      FlutterSemanticsFlag::kFlutterSemanticsFlagIsChecked);
+  bridge->AddFlutterSemanticsNodeUpdate(&root);
+
+  bridge->CommitUpdates();
+
+  auto root_node = bridge->GetFlutterPlatformNodeDelegateFromID(0).lock();
+  EXPECT_EQ(root_node->GetData().role, ax::mojom::Role::kCheckBox);
+  EXPECT_EQ(root_node->GetData().GetCheckedState(), ax::mojom::CheckedState::kTrue);
+}
```

Comment:
Looks good so far! Probably worth adding an update that verifies that sending a node update for the root node to an unchecked state.

If you want to check the IAccessible checked state, you can add something like the following, which I haven't tested myself, but should be roughly what you want:
```cc
// Get the IAccessible for the root node.
IAccessible* native_view = root_node->GetNativeViewAccessible();
ASSERT_TRUE(native_view != nullptr);

// Look up against the node itself (not one of its children)
VARIANT varchild = {};
varchild.vt = VT_I4;

// Verify the checkbox is checked.
varchild.lVal = CHILDID_SELF;
VARIANT native_state = {};
ASSERT_TRUE(SUCCESSFUL(native_view->get_accState(varchild, &native_state)));
ASSERT_EQ(native_state.lVal, VT_I4);
EXPECT_TRUE(native_state & STATE_SYSTEM_CHECKED);
```

Reply:
Maybe you can help me deal with this. `GetNativeViewAccessible` is returning `nullptr`, and I do not know why this would be. I have not had any previous exposure to most of the methods, types, and macros, so I do not know if that is correct or how to get around this problem

Reply:
Discussed over chat.